docs/gitbook/architecture/topology/patterns.adoc

```diff
@@ -0,0 +1,164 @@
+
+image:img/hub_and_connected_spoke.svg["The Hub and Connected Spoke
+molecule", 175, 194, role="right"]
+The Hub and Connected spoke molecule has at least two atom types and is
+a more versatile version of the Hub and Spoke molecule. There is a
+greater potential for throughput increase, as some processing is
+offloaded from the hub onto the connected spokes. Privacy of spokes is
+assured, as sensitive information is passed directly to the concerned
+spokes.
+
+In this molecule, the hub could act as a directory blockchain providing
+public details about all of the connected spokes. When spokes want to
+communicate privately, they can create and connect to an intermediate
+blockchain that allows them to share data that neither the hub nor the
+other spokes could know about or access.
+
+
+=== Control by Proxy
+
+image:img/control_by_proxy.svg["The Control By Proxy molecule", 175,
+194, role="right"]
+The Control by Proxy molecule is a more-versatile version of a Hub and
+Spoke molecule that uses at least three atom types. One of the spokes is
+of a different type -- the _control_ atom (in red). The hub (in blue)
+acts as an intermediary. The spokes (in green) may not see any part of
+the control atom, and the control atom can have additional permissions
+in the hub that the spokes would not. As a result, the most sensitive
+data can be placed in the control atom without worry.
+
+This topology is useful, for example, for
+link:https://en.wikipedia.org/wiki/Health_Insurance_Portability_and_Accountability_Act[HIPAA]
+compliance. Imagine that the control atom is managed by a hospital.
+Patient information could flow to the spokes, each of which would be
+managed by an insurance company. The hospital would know who the
+patients are, and the insurance companies need a reliable identifier for
+each patient. The hub would be managed by a compliance organization,
+providing the persistent association of the hospital patients with their
+"anonymous" insurance identifiers.
+
+
+== Combinations
+
+=== The Multi-Tier Hub and Spoke Molecule
+
+image:img/hierarchical_throughput.svg["The Multi-tier Hub and Spoke
+Model", 175, 194, role="right"]
+Using the Hub and Spoke molecule may present a performance problem if
+one or more spokes needs to dispatch more actions than the hub can
+handle. For example, in a blockchain network for a banking system,
+perhaps one branch in a bank is in the dense city center and processes
+triple the transactions of the other branches.
+
+The Multi-Tier Hub and Spoke molecule would be a good solution:
+the busy branch could be split into its own hub and spoke molecule,
+such as one blockchain for the branch itself and separate blockchains
+for each automated teller machine (ATM). The branch could then aggregate
+its data upwards to the hub at regular intervals. The splitting into
+multiple blockchains distributes the processing load -- provided that
+the new blockchains are hosted on separate nodes from the original node.
```

Comment:
It may not even be necessary to run these on different nodes as it's likely the time to consensus is the limiter, not the raw processing power of the nodes.